src/plumtree_broadcast.erl

```diff
@@ -568,10 +568,33 @@ send(Msg, Peers) when is_list(Peers) ->
     [send(Msg, P) || P <- Peers];
 send(Msg, P) ->
     %% TODO: add debug logging
-    gen_server:cast({?SERVER, P}, Msg).
+    gen_server2:cast({?SERVER, P}, Msg).
 
 schedule_lazy_tick() ->
-    schedule_tick(lazy_tick, broadcast_lazy_timer, 1000).
+    schedule_lazy_tick(0, os:timestamp()).
+schedule_lazy_tick(I_last, T_last) ->
+    %% Schedules the next lazy tick. Sticking to a fixed time interval causes
+    %% an exponential growth of produced i_have messages if this process is
+    %% overloaded (process mailbox is huge) and has many outstanding lazy messages.
+    %%
+    %% The idea is to measure the time it takes the lazy tick message to traverse
+    %% the mailbox and increase the next tick time if the process is overloaded.
+    %% The simple power function is used:
+    %%
+    %%      f(x): I * (1 + x)^C
+    %%
+    %%      x: overhead to traverse the mailbox
+    %%      I: constant: min tick interval, ensures the lower bound
+    %%      C: constant: 0 =< C =< 1
+    %%
+    %%      x = t_now - t_last - i_last | t_now >= t_last
+    %%
+    I = app_helper:get_env(plumtree, min_lazy_tick_interval, 10000),
+    C = 0.25,
+    T_now = os:timestamp(),
+    X = abs((timer:now_diff(T_now, T_last) div 1000) - I_last),
+    Y = round(I * math:pow(1 + X, C)),
+    schedule_tick({lazy_tick, Y, T_now}, broadcast_lazy_timer, Y).
```

Comment:
Would it maybe be nice to log something (info level) if we are scheduling a tick with a longer interval than `I` ?